pkg/controller/checker/checker_test.go

```diff
@@ -172,83 +216,114 @@ func TestContainer(t *testing.T) {
 				ImageURL:       "localhost:5000/version-checker",
 				IsLatest:       true,
+				OS:             linux,
+				Architecture:   amd64,
 			},
 		},
 		"if using sha but not latest, return false": {
 			statusSHA: "localhost:5000/version-checker@sha:123",
 			imageURL:  "localhost:5000/joshvanl/version-checker@sha:123",
 			opts:      new(api.Options),
 			searchResp: &api.ImageTag{
-				Tag: "v0.2.0",
-				SHA: "sha:456",
+				Tag:          "v0.2.0",
+				SHA:          "sha:456",
+				OS:           linux,
+				Architecture: amd64,
 			},
 			expResult: &Result{
 				CurrentVersion: "sha:123",
 				LatestVersion:  "v0.2.0@sha:456",
 				ImageURL:       "localhost:5000/joshvanl/version-checker",
 				IsLatest:       false,
+				OS:             linux,
+				Architecture:   amd64,
 			},
 		},
 		"if using sha but sha not latest, return false and no tag if non exists": {
 			statusSHA: "localhost:5000/version-checker@sha:123",
 			imageURL:  "localhost:5000/joshvanl/version-checker@sha:123",
 			opts:      new(api.Options),
 			searchResp: &api.ImageTag{
-				Tag: "",
-				SHA: "sha:456",
+				Tag:          "",
+				SHA:          "sha:456",
+				OS:           linux,
+				Architecture: amd64,
 			},
 			expResult: &Result{
 				CurrentVersion: "sha:123",
 				LatestVersion:  "sha:456",
 				ImageURL:       "localhost:5000/joshvanl/version-checker",
 				IsLatest:       false,
+				OS:             linux,
+				Architecture:   amd64,
 			},
 		},
 		"if using sha and is latest, return true": {
 			statusSHA: "localhost:5000/version-checker@sha:123",
 			imageURL:  "localhost:5000/joshvanl/version-checker@sha:123",
 			opts:      new(api.Options),
 			searchResp: &api.ImageTag{
-				Tag: "v0.2.0",
-				SHA: "sha:123",
+				Tag:          "v0.2.0",
+				SHA:          "sha:123",
+				OS:           linux,
+				Architecture: amd64,
 			},
 			expResult: &Result{
 				CurrentVersion: "sha:123",
 				LatestVersion:  "v0.2.0@sha:123",
 				ImageURL:       "localhost:5000/joshvanl/version-checker",
 				IsLatest:       true,
+				OS:             linux,
+				Architecture:   amd64,
 			},
 		},
 		"if using sha and is latest, return true and no tag if non exists": {
 			statusSHA: "localhost:5000/version-checker@sha:123",
 			imageURL:  "localhost:5000/joshvanl/version-checker@sha:123",
 			opts:      new(api.Options),
 			searchResp: &api.ImageTag{
-				Tag: "",
-				SHA: "sha:123",
+				Tag:          "",
+				SHA:          "sha:123",
+				OS:           linux,
+				Architecture: amd64,
 			},
 			expResult: &Result{
 				CurrentVersion: "sha:123",
 				LatestVersion:  "sha:123",
 				ImageURL:       "localhost:5000/joshvanl/version-checker",
 				IsLatest:       true,
+				OS:             linux,
+				Architecture:   amd64,
 			},
 		},
 	}
 
+	architectureMap := architecture.New()
+	node := &corev1.Node{}
+	node.Name = "node1"
+	node.Labels = map[string]string{
+		"kubernetes.io/os":   linux,
+		"kubernetes.io/arch": amd64,
+	}
+	architectureMap.AddNode(node)
```

Comment:
Move this existing node into the test struct input. We should test for when a node does and doesn't exist on the test input.

```go
	tests := map[string]struct {
		statusSHA  string
		imageURL   string
		opts       *api.Options
    nodes      []*corev1.Node
		searchResp *api.ImageTag
		expResult  *Result
	}{
```

```go

nodes: []*corev1.Node {
{  
&corev1.Node {
  ObjectMeta: &metav1.ObjectMeta {
    Name: "node1"
    Labels: map[string]string{
    corev1.LabelArchStable: archLinux
   },
  },
},
````
etc.